datafusion/core/src/physical_optimizer/output_requirements.rs

```diff
@@ -248,7 +248,9 @@ fn require_top_ordering_helper(
     if children.len() != 1 {
         Ok((plan, false))
     } else if let Some(sort_exec) = plan.as_any().downcast_ref::<SortExec>() {
-        let req_ordering = sort_exec.properties().output_ordering().unwrap_or(&[]);
+        // In the case of constant columns, output ordering of SortExec would give an emtpy set.
+        // Therefore; we check the sort expression field of the SortExec to assign the requirements.
```

Comment:
```suggestion
        // In case of constant columns, output ordering of SortExec would give an empty set.
        // Therefore; we check the sort expression field of the SortExec to assign the requirements.
```